akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -254,8 +254,12 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
    * @param events events to be persisted
    * @param handler handler for each persisted `events`
    */
+  final def persistAll[A](events: immutable.Seq[A])(handler: A ⇒ Unit): Unit =
+    events.foreach(persist(_)(handler)) // TODO the atomic part should be handled by issue #15377
+
+  @deprecated("use persistAll instead", "2.4")
   final def persist[A](events: immutable.Seq[A])(handler: A ⇒ Unit): Unit =
-    events.foreach(persist(_)(handler))
+    persistAll(events)(handler)
```

Comment:
We must keep the deprecated for 2.4. We can't rely on that users migrate via milestones. It would be the worst kind of breaking change that we can do.